### DISCUSSION TITLE
Is there an easy way to disable thread parallelism for tests in one file?

### DISCUSSION THREAD

**Discussion by yurivict** [pytest-dev/pytest#12188] [General]
I have a file with several dozen test_xx functions.

They appear to be run in parallel by PyTest, and since one underlying library isn't thread safe - these tests fail intermittently.

One way to fix this is to add thread locking into each of test_xx procedures, but this would be a lot of changes.

Is there a way to set an option instructing PyTest to run these tests sequentially?

**Comment by The-Compiler** [comment]
pytest doesn't run anything in parallel. There's the pytest-xdist plugin that allows doing so, but that uses processes and not threads.

**Comment by mandras73** [comment]
take a look on the docs [here](https://pytest-xdist.readthedocs.io/en/stable/distribution.html)
you may use `--dist loadgroup`
and set test a unique group marker if you would like to execute that one in sequential
example:

```
@pytest.mark.xdist_group(name="group1")
def test1():
    pass

def test2():
    pass

def test3():
    pass
```

Results: test2 and test3 run parallel, test1 runs sequential (in a separated group)

**Comment by RonnyPfannschmidt** [comment]
Closing this as pytest is not using threads and no details suggesting their origin was provided

**Comment by calebcpAtAmazon** [comment]
If you're using the pytest-xdist plugin (which you are probably are if you're on this discussion), you can disable parallelism by specifying `-n 0` when you run pytest. For example, if you're running pytest against "src" folder, you would run:

```
pytest src -n 0
```
